version: '3.8'

services:
  postgres:
    image: postgres:16-alpine
    container_name: postgres-pgvector
    networks:
      - selvanet
    entrypoint: ["docker-entrypoint.sh"]
    command: ["postgres", "--max_connections=200"]
    ports:
      - "5432:5432"
    volumes:
      - postgres-data:/var/lib/postgresql/data
      - ./pgvector-init.sh:/docker-entrypoint-initdb.d/init-vector.sh
    environment:
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_INITDB_ARGS: "--auth-host=scram-sha-256"
      POSTGRES_USER: ${POSTGRES_USER:-postgres}
      POSTGRES_DB: ${POSTGRES_DB:-postgres}
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 5
    deploy:
      mode: replicated
      replicas: 1
      placement:
        constraints:
          - node.role == manager
      resources:
        limits:
          cpus: "0.5"
          memory: 256M

networks:
  selvanet:
    external: true

volumes:
  postgres-data:
    driver: local
